test/framework/robot.py

```diff
@@ -567,6 +567,14 @@ def test_get_toolchain_hierarchy(self):
             {'name': 'gompi', 'version': '1.4.10'},
```

Comment:
Wait, this problem is also the cause of the second test, it must be related to the change you made in the toolchain hierarchy generator